src/test/java/it/pagopa/pagopa/apiconfig/service/IcaServiceTest.java

```diff
@@ -158,4 +169,88 @@ void deleteIca() {
+    }
+
+    // xsd not valid
+    @Test
+    void verifyIca_ko_1() throws IOException {
+        File xml = TestUtil.readFile("file/ica_xsd_not_valid.xml");
+        MockMultipartFile file = new MockMultipartFile("file", xml.getName(), MediaType.MULTIPART_FORM_DATA_VALUE, new FileInputStream(xml));
+
+        List<CheckItem> checkItemList = icaService.verifyIca(file);
+        assertEquals(1, checkItemList.stream().filter(item -> item.getValid().equals(CheckItem.Validity.NOT_VALID)).count());
+    }
+
+    // creditor institution not consistent
+    // validityDate not consistent
+    @Test
+    void verifyIca_ko_2() throws IOException {
+        File xml = TestUtil.readFile("file/ica_info_not_valid_h2.xml");
+        MockMultipartFile file = new MockMultipartFile("file", xml.getName(), MediaType.MULTIPART_FORM_DATA_VALUE, new FileInputStream(xml));
+        when(paRepository.findByIdDominio(anyString())).thenReturn(Optional.empty());
+
+        List<CheckItem> checkItemList = icaService.verifyIca(file);
+        assertEquals(2,checkItemList.stream().filter(item -> item.getValid().equals(CheckItem.Validity.NOT_VALID)).count());
+    }
+
+
+    // qr-code not present
+    // flow ìdentifier present
+    @Test
+    void verifyIca_ko_3() throws IOException {
+        File xml = TestUtil.readFile("file/ica_info_not_valid_h2.xml");
+        MockMultipartFile file = new MockMultipartFile("file", xml.getName(), MediaType.MULTIPART_FORM_DATA_VALUE, new FileInputStream(xml));
+        when(paRepository.findByIdDominio(anyString())).thenReturn(Optional.of(getMockPa()));
+
+        CodifichePa barcode128aim = CodifichePa.builder()
+                .id(1L)
+                .codicePa("000000001016")
+                .fkCodifica(Codifiche.builder()
+                        .objId(2L)
+                        .idCodifica(Encoding.CodeTypeEnum.BARCODE_128_AIM.getValue())
+                        .build())
+                .build();
+
+        when(codifichePaRepository.findAllByFkPa_ObjId(anyLong())).thenReturn(List.of(barcode128aim));
+        when(informativeContoAccreditoMasterRepository.findByIdInformativaContoAccreditoPaAndFkPa_IdDominio(anyString(), anyString())).thenReturn(Optional.of(getMockInformativeContoAccreditoMaster()));
+
+        IbanValidiPerPa iban_1 = IbanValidiPerPa.builder()
+                .ibanAccredito("IT04I0103061821000000248378")
+                .build();
+
+        IbanValidiPerPa iban_2 = IbanValidiPerPa.builder()
+                .ibanAccredito("IT45R0760103200000000001016")
+                .build();
+
+        List<IbanValidiPerPa> ibans = List.of(iban_1, iban_2);
+
+        when(ibanValidiPerPaRepository.findAllByFkPa(anyLong())).thenReturn(ibans);
+
+        List<CheckItem> checkItemList = icaService.verifyIca(file);
+        assertEquals(3,checkItemList.stream().filter(item -> item.getValid().equals(CheckItem.Validity.NOT_VALID)).count());
+    }
+
+    //
+//    @Test
```

Comment:
why are some tests commented? Can we remove them?

Reply:
cut & paste 😄